sup3r/preprocessing/data_handling/nc_data_handling.py

```diff
@@ -685,16 +682,20 @@ def get_clearsky_ghi(self):
                                                 self._nsrdb_smoothing,
                                                 mode='nearest')
 
-        if cs_ghi.shape[-1] < t_end_target:
-            n = int(np.ceil(t_end_target / cs_ghi.shape[-1]))
+        if cs_ghi.shape[-1] < len(self.time_index):
+            n = int(np.ceil(len(self.time_index) / cs_ghi.shape[-1]))
             cs_ghi = np.repeat(cs_ghi, n, axis=2)
-            cs_ghi = cs_ghi[..., :t_end_target]
+            cs_ghi = cs_ghi[..., :len(self.time_index)]
 
         logger.info(
             'Reshaped clearsky_ghi data to final shape {} to '
             'correspond with CC daily average data over source '
             'temporal_slice {} with (lat, lon) grid shape of {}'.format(
                 cs_ghi.shape, self.temporal_slice, self.grid_shape))
+        
+
+        msg = ('cs_ghi time dimension ({}) does not match input time dimension ({})'.format(cs_ghi.shape[2], len(self.time_index)))
```

Comment:
This should be "nsrdb clearsky GHI time dimension {} does not match the GCM time dimension {}"